File: NextApp/iOS/OrgPortalNext/Core/DataLayer/FriendExchangeRepository.swift
import Foundation
import Model

@MainActor
public protocol FriendExchangeRepository: AnyObject {
    func fetchContacts() async throws -> [FriendContact]
    func fetchHistories(friendId: UUID) async throws -> [FriendInteractionHistory]
    func save(_ contact: FriendContact) async throws
    func save(_ history: FriendInteractionHistory) async throws
    func deleteContact(id: UUID) async throws
    func deleteHistory(id: UUID) async throws
}

File: NextApp/Android/core/data/src/main/kotlin/jp/komehyappyo/member/next/core/data/FriendExchangeRepository.kt
package jp.komehyappyo.member.next.core.data

import jp.komehyappyo.member.next.core.model.FriendContact
import jp.komehyappyo.member.next.core.model.FriendInteractionHistory
import kotlinx.coroutines.flow.Flow
import java.util.UUID

interface FriendExchangeRepository {
    fun observeContacts(): Flow<List<FriendContact>>
    fun observeHistories(friendId: UUID): Flow<List<FriendInteractionHistory>>
    suspend fun save(contact: FriendContact)
    suspend fun save(history: FriendInteractionHistory)
    suspend fun deleteContact(id: UUID)
    suspend fun deleteHistory(id: UUID)
}
